### DISCUSSION TITLE
Comments regardin schema documentation

### DISCUSSION THREAD

**Discussion by dethodav** [gwosc-tutorial/gwosc-catalog#72] [General]
I've been going through the schema documentation provided here and have a number of suggestions questions that came up while trying to follow the schema to create a mock catalog file. Overall, it was fairly straightforward to follow the schema, but there were a number of details that should be clarified. 

* The definition of `detectors` is not clear (Currently "A list of detector data used for this event"). Specifically, the choice to have this be an event-level parameter and the use of the term "used" in the definition. For example, does this refer to what detectors were in observing mode? The superset of all detectors that were used in any analysis in the catalog, even if not in observing mode? In addition to clarifying this definition, I suggest adding detectors as a parameter that can be provided per search/pe result
* Related to #57, it is not clear what fields are required/allowed/optional throughout the schema. It would be helpful to reframe these requirements based on what GWOSC will do with the data. What is the minimum set of fields that are needed to process the catalog, what are the fields that will be shown on the GWOSC website, and what fields will be accessible via some other method if provided? 
* Related to #70, how is the `decimal_places` field used? If I provide a value for a parameter that is different from the number of decimal places than is listed in this field, what will happen?
* Related to the above, is it possible to use `decimal_places` to indicate how many significant figures should be used?
* I'm not sure how the `is_upper_bound`/`is_lower_bound` fields interplay with the `median`/`lower_5`/`upper_95` values. What is "this value" referring to in the bound definitions? I can imagine different definitions of these fields depending on whether credible intervals are being quoted or point-estimates. 
* Related to the above, does `median` have to be a median?  
* Does `waveform_family` need to be anything specific? If I don't use a single specific waveform approximant in my analysis, can I use this field for something else? 
* The definition of `snr` in the search parameter_name section states that this must be from matched filtering, but not all GW searches use matched filtering
* Similarly, what if a search doesn't report significance using a false-alarm rates with units of `1/time` or doesn't report a p-astro?
* Should "confidence" interval be "credible" interval? 
* The phrase "PE standard names used by ligo" should be "definition of parameters commonly used in PE by the LVK collaboration"
* The link provided with "A list of waveform family names" should be made more specific - I don't know where I specifically can find this list in the linked documentation
* The phrase "A list of waveform family names" likely should be "Examples of waveform family names that have been produced for gravitational-wave data analysis" (since any linked list is unlikely to be exhaustive)

**Comment by martinberoiz** [comment]
Hi Derek, thanks for the comments! I think I will resume working on this repo in a few weeks from now, after I finish some other stuff I have. This is very helpful.

Also, I'm moving this to the Discussions section and I'll be opening more actionable issues coming from these points you mention.

**Reply by DrKentB** [reply]
Hi Derek, Martin, here is a pass at some of your bullets in order:

- definition of detector: in my opinion this is the list of detectors in observing mode at the time of the event. on rare occasions a detector not in observing mode has been promoted for use in an event in which case it may be added if data from that detector did in fact get used. (Hoping this becomes less likely going forward).
- what fields are required: based on the work done by the IAS group, I think the intention was for all fields to be provided, and if there is no mapping of the field, then to specify "none". Having said this on the bases of the IAS work, I think the schema has evolved while I was out to support omitting fields and to have default values applied in the cases of is_upper_bound and is_lower_bound. The only other key word I can find in the documentation that is optional is is_preferred. I think there is an oversight in the documentation regarding median, upper_95 and lower_95 ... my interpretation is that if the parameter has a name, i.e., parameter_name is specified then the median should be specified, and if it comes from a distribution, then upper_95 and lower_95 should be specified (but not required). I think median is too strict an interpretation of what goes here ... more like expectation value when the number is not drawn from a distribution.
- decimal place field: the number in the data base can have arbitrary precision, but how it is displayed in GWOSC in a table will be limited to the number of decimal places specified here. This is not significant figures and it is not possible to explicitely express sigfigs in the GWOSC, but as a poor-person's tool, this will give some control.
- decimal places does not capture significant figures and there is no way in GWOSC to capture this (yes I've complained about this myself in the past, but with a bit of manual intervention it has been able to get the wanted value across so far)
- Interplay of is_upper_bound / is_lower_bound: PE results have been published with these and this is how they are captured in the database. The same "object" is used for other values such as search results where the values do not have a reported upper and lower bound. if there is an upper_95 and/or lower_95 then set is _upper_bound and is_lower_bound to true, otherwise set it to false. I think naming these "has_upper_bound" and "has_lower_bound" would make more sense - my two cents.
- Median does not have to be a median, it can be the expected value when not drawn from a distribution
- waveform family can be anything you like. but historically that value is found in the hdf5 samples file (and more often than not is reported as "c01:mixed". This gives the GWOSC users a mapping of the waveform documented in the hdf5 file.
- Regarding SNR, I think the intent is to distinguish this from other SNR's that may be found by, e.g., PE for an apples to apples comparison when sorting on this column in gwosc. Of course there are so many ways to calculate SNR that ultimately any documentation (the published paper) should expose what is meant by SNR
- Similar to above, FAR is captured in the GWOSC database with all values using the same units (1/yr in GWOSC) so that that column can be sorted. For Pastro, there are causes where the event didn't have Pastro so it is not required to be there, but does add useful content.
- I agree that this should be "definition of parameters commonly used in PE by the LVK collaboration". I think there is a links to a document on this (https://lscsoft.docs.ligo.org/pesummary/stable/gw/parameters.html) that might be public.
- regarding (A list of waveform family names can be found in the lalsuite code site https://lscsoft.docs.ligo.org/lalsuite/.), I don't know why this is here. I believe the waveform family is defined by the analyst and should be flexible, though documented somehow, either in the paper or online gwosc documentation for the catalog.
- last bullet - I think I covered this in the previous bullet.

I would like to add one thing -  I don't think values of NaN make since and instead just saying None would more realistic for any published result.
